DQMOffline/Alignment/src/TkAlCaRecoMonitor.cc

```diff
@@ -222,15 +224,14 @@ void TkAlCaRecoMonitor::analyze(const edm::Event &iEvent, const edm::EventSetup
   AlCaRecoTrackEfficiency_->Fill(static_cast<double>((*trackCollection).size()) / (*referenceTrackCollection).size());
 
   double sumOfCharges = 0;
-  for (reco::TrackCollection::const_iterator track = (*trackCollection).begin(); track < (*trackCollection).end();
-       ++track) {
+  for (const auto &track : *trackCollection) {
     double dR = 0;
     if (runsOnReco_) {
       double minJetDeltaR = 10;  // some number > 2pi
-      for (reco::CaloJetCollection::const_iterator itJet = jets->begin(); itJet != jets->end(); ++itJet) {
-        jetPt_->Fill((*itJet).pt());
-        dR = deltaR((*track), (*itJet));
-        if ((*itJet).pt() > maxJetPt_ && dR < minJetDeltaR)
+      for (const auto &itJet : *jets) {
+        jetPt_->Fill(itJet.pt());
+        dR = deltaR(track, itJet);
```

Comment:
Minor point: `deltaR2` is computationally less expensive

Reply:
but then isn't taking the root moving the expense elsewhere (notice we are interested in dR, not dR2). Or are you suggesting to plot directly dR2?

Reply:
As I said, it is a minor point. But yes, computing the square root only once before filling the histograms is less expensive than having to compute it at every jet or track

Reply:
sorry, I admit I misunderstood what you proposed. Here's the requested improvement: https://github.com/cms-sw/cmssw/pull/40887.